src/main/scala/nl/knaw/dans/easy/stage/dataset/Util.scala

```diff
@@ -67,6 +67,25 @@ object Util {
     } yield audience.text
   }
 
+  def readFileType(filePath: String)(implicit s: Settings): Try[Option[String]] = Try {
+    val filetype = for {
+      file <- loadBagXML("metadata/files.xml") \\ "files" \ "file"
+      if (file \ "@filepath").text == filePath
+      filetype <- file \ "type"
+    } yield filetype
+    if (filetype.size == 1) Option(filetype.head.text)
+    else None
```

Comment:
So, what you're doing here is going through the whole list to find a certain element and when you found it, you continue iterating the whole list until you reached the end. Then you collect all the results in a list and check whether the list contains any elements, after which you return the head of the list in a `Some`; otherwise returning `None`.

You can improve on this by not using a for-comprehension here, but a `find` instead. It'll be something like

```scala
(loadBagXML("metadata/files.xml") \\ "files" \ "file")
  .find(file => (file \@ "filepath").text == filePath) // searches for the first element that satisfies the predicate and returns it in a `Some`; if no elements satisfy, it returns a `None`
  .map(file => file \ "type")
```